packages/react/src/prefabs/PreJoin.tsx

```diff
@@ -87,31 +121,33 @@ export function usePreviewDevice<T extends LocalVideoTrack | LocalAudioTrack>(
   };
 
   const switchDevice = async (track: LocalVideoTrack | LocalAudioTrack, id: string) => {
-    await track.restartTrack({
-      deviceId: id,
-    });
+    await track.setDeviceId(id);
     prevDeviceId.current = id;
   };
 
   const prevDeviceId = React.useRef(localDeviceId);
 
   React.useEffect(() => {
-    if (enabled && !localTrack && !deviceError) {
+    if (enabled && !localTrack && !deviceError && !isCreatingTrack) {
       log.debug('creating track', kind);
-      createTrack(localDeviceId, kind);
+      setIsCreatingTrack(true);
+      createTrack(localDeviceId, kind).finally(() => {
+        setIsCreatingTrack(false);
+      });
     }
-  }, [enabled, localTrack, deviceError]);
+  }, [enabled, localTrack, deviceError, isCreatingTrack]);
 
   // switch camera device
   React.useEffect(() => {
+    if (!localTrack) {
+      return;
+    }
     if (!enabled) {
       if (localTrack) {
```

Comment:
this is redundant with the previous return on `!localTrack` ?

Reply:
true! thanks